microprofile/telemetry/src/main/java/io/helidon/microprofile/telemetry/HelidonTelemetryContainerFilter.java

```diff
@@ -0,0 +1,186 @@
+import jakarta.ws.rs.container.ContainerResponseFilter;
+import jakarta.ws.rs.container.ResourceInfo;
+import jakarta.ws.rs.ext.Provider;
+
+import static java.net.HttpURLConnection.HTTP_OK;
+
+/**
+ * Filter to process Server request and Server response. Starts a new {@link io.opentelemetry.api.trace.Span} on request and
+ * ends it on a Response.
+ */
+@Provider
+class HelidonTelemetryContainerFilter implements ContainerRequestFilter, ContainerResponseFilter {
+    private static final System.Logger LOGGER = System.getLogger(HelidonTelemetryContainerFilter.class.getName());
+    private static final String REST_RESOURCE_CLASS = "rest.resource.class";
+    private static final String REST_RESOURCE_METHOD = "rest.resource.method";
+    private static final String SPAN = "otel.span.server.span";
+    private static final String SPAN_CONTEXT = "otel.span.server.context";
+    private static final String SPAN_SCOPE = "otel.span.server.scope";
+    private static final String HTTP_STATUS_CODE = "http.status_code";
+    private static final String HTTP_METHOD = "http.method";
+    private static final String HTTP_SCHEME = "http.scheme";
+    private static final String HTTP_TARGET = "http.target";
+
+    // Extract OpenTelemetry Parent Context from Request headers.
+    private static final TextMapGetter<ContainerRequestContext> CONTEXT_HEADER_INJECTOR;
+    private final Tracer tracer;
+    private final OpenTelemetry openTelemetry;
+
+    static {
+        CONTEXT_HEADER_INJECTOR =
+                new TextMapGetter<>() {
+                    @Override
+                    public String get(ContainerRequestContext containerRequestContext, String s) {
+                        Objects.requireNonNull(containerRequestContext);
+                        return containerRequestContext.getHeaderString(s);
+                    }
+
+                    @Override
+                    public Iterable<String> keys(ContainerRequestContext containerRequestContext) {
+                        return List.copyOf(containerRequestContext.getHeaders().keySet());
+                    }
+                };
+    }
+
+    @jakarta.ws.rs.core.Context
+    private ResourceInfo resourceInfo;
+
+    @Inject
+    HelidonTelemetryContainerFilter(Tracer tracer, OpenTelemetry openTelemetry) {
+        this.tracer = tracer;
+        this.openTelemetry = openTelemetry;
+    }
+
+    @Override
+    public void filter(ContainerRequestContext requestContext) {
+
+        if (LOGGER.isLoggable(System.Logger.Level.TRACE)) {
+            LOGGER.log(System.Logger.Level.TRACE, "Starting Span in a Container Request");
+        }
+
+        Context parentContext = Context.current();
+
+        // Extract Parent Context from request headers to be used from previous filters
+        Context extractedContext = openTelemetry.getPropagators().getTextMapPropagator()
+                .extract(Context.current(), requestContext, CONTEXT_HEADER_INJECTOR);
```

Comment:
There is `Context.current()` called twice. Should be probably merged to one call, if I am not mistaken by something I do not see.

Reply:
I have already mentioned that in the previous review and it was marked as resolved.

Reply:
No, it should be done this way, as  during the calls the result of `Context.current()` changes.